### PR TITLE
Unit Test for Controllers and Models

### PR DESCRIPTION
**Pull request by nlperez11**
## Summary
- Unit test for BookController
- Unit test for UserController
- Override toString for Book Model
- Unit test for User and Book model
## Trello Card
https://trello.com/c/WfOaZbtp/28-junit-testing

### CODE REVIEW

File: src/test/java/com/wolox/training/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doNothing;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
Comment (martinremis) [inline src/test/java/com/wolox/training/controller/BookControllerTest.java]: Try not to use wildcards at imports.

File: src/test/java/com/wolox/training/controller/UserControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doNothing;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
Comment (martinremis) [inline src/test/java/com/wolox/training/controller/UserControllerTest.java]: Same

**Review by martinremis — CHANGES_REQUESTED** [review]
Good job! Just a comment...

File: src/test/java/com/wolox/training/controller/BookControllerTest.java (hunk 1 above)
Comment (nlperez11) [reply src/test/java/com/wolox/training/controller/BookControllerTest.java]: Solved, i've replaced the wildcard for specific methods

File: src/test/java/com/wolox/training/controller/UserControllerTest.java (hunk 2 above)
Comment (nlperez11) [reply src/test/java/com/wolox/training/controller/UserControllerTest.java]: Solved, i've replaced the wildcard for specific methods

**Review by martinremis — APPROVED** [review]

**Review by jeymarsonwolox — APPROVED** [review]

**Pull request merged**